examples/ui5-js-app/webapp/test/e2e/evalOnControl.test.js

```diff
@@ -0,0 +1,112 @@
+const Main = require("./pageObjects/Main")
+const Other = require("./pageObjects/Other")
+const marky = require("marky")
+const { wdi5 } = require("wdio-ui5-service")
+
+describe("ui5 eval on control", () => {
+    before(async () => {
+        await Main.open()
+    })
+
+    it("should have the right title", async () => {
+        const title = await browser.getTitle()
+        expect(title).toEqual("Sample UI5 Application")
+    })
+
+    it("execute function browserside on button to get its text, basic return type", async () => {
+        const button = await browser.asControl({
+            selector: {
+                id: "openDialogButton",
+                viewName: "test.Sample.view.Main"
+            }
+        })
+        const buttonText = await button.evalOnControl(function () {
+            return this.getText()
+        })
+        expect(buttonText).toEqual("open Dialog")
+
+        const regularBtnText = await button.getText()
+        expect(buttonText).toEqual(regularBtnText)
+    })
+
+    it("nav to other view and get people list names, array return type", async () => {
+        // click webcomponent button to trigger navigation
+        const navButton = await browser.asControl({
+            selector: {
+                id: "NavFwdButton",
+                viewName: "test.Sample.view.Main"
+            }
+        })
+        await navButton.press()
+
+        const listSelector = {
+            selector: {
+                id: "PeopleList",
+                viewName: "test.Sample.view.Other",
+                interaction: "root"
+            }
+        }
+        const list = await browser.asControl(listSelector)
+
+        /**
+         * need to set
+         * wdi5: {logLevel: "verbose"}
+         * in config.js
+         */
+
+        // *********
+        // new approach -> takes ~4.3sec
+        marky.mark("evalOnControlForListItemTitles")
+        const peopleListNames = await list.evalOnControl(function () {
```

Comment:
similar to my suggestion above:
```js
await list.evalOnControl( getItems().map( item => item.getTitle() )
```

Reply:
this would even allow to dynamically change the return type, à la
```js
// have an object returned with an embedded array
await list.evalOnControl(  { getItems().map( item => item.getTitle() } )
```